### PR TITLE
DM-46885: Set up a Prompt Processing deployment configured for ComCam.

### PR DESCRIPTION
**Pull request by erinleighh**

### CODE REVIEW

File: applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        (survey="cwfs-focus-sweep")=[]
+        (survey="spec-survey")=[]
+        (survey="BLOCK-271")=[]
+        (survey="BLOCK-295")=[]
+        (survey="")=[]
+    calibRepo: s3://rubin-summit-users
+
+  s3:
+    imageBucket: rubin-summit
+    endpointUrl: https://s3dfrgw.slac.stanford.edu
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml:49]: `https://sdfembs3.sdf.slac.stanford.edu`

File: applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  raw_microservice: http://172.24.5.144:8080/presence
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml:51]: 172.24.5.158

File: applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  imageNotifications:
+    kafkaClusterAddress: prompt-processing-2-kafka-bootstrap.kafka:9092
+    topic: rubin-prompt-processing-prod
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml:55]: rubin-summit-notification

File: applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    tag: 4.5.1
+
+  instrument:
+    pipelines:
+      # BLOCK-306 is photographic imaging
+      # BLOCK-T17 is daytime checkout
+      # BLOCK-271 is photon transfer curve calibrations
+      # BLOCK-295 is the daily calibration sequence as of May 27, 2024
+      main: >-
+        (survey="BLOCK-306")=[${PROMPT_PROCESSING_DIR}/pipelines/LSSTComCam/ApPipe.yaml,
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml:26]: These survey names are for LATISS and are unlikely to be the same for LSSTComCam. See the [Slack thread ](https://rubin-obs.slack.com/archives/C07QJMQC8KC/p1729703256595719). It's okay to put in a simpler placeholder for now.

File: applications/prompt-proto-service-lsstcomcam/README.md [hunk 5]
```diff
@@ -15,11 +15,11 @@ Prompt Proto Service is an event driven service for processing camera images. Th
 | prompt-proto-service.additionalVolumeMounts | list | `[]` | Kubernetes YAML configs for extra container volume(s). Any volumes required by other config options are automatically handled by the Helm chart. |
 | prompt-proto-service.affinity | object | `{}` | Affinity rules for the prompt processing pods |
 | prompt-proto-service.alerts.server | string | `"usdf-alert-stream-dev-broker-0.lsst.cloud:9094"` | Server address for the alert stream |
-| prompt-proto-service.alerts.topic | string | None, must be set | Topic name where alerts will be sent |
+| prompt-proto-service.alerts.topic | string | `""` | Topic name where alerts will be sent |
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/README.md:18]: This commit of doc update should be squashed with the commit where you update the values

File: applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    revision: "1"
+
+  # HACK: disable autoscaling as workaround for DM-41829
+  worker:
+    restart: 15
+
+  image:
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: 4.5.1
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values-usdfprod-prompt-processing.yaml:17]: `4.6.0` ?

File: applications/prompt-proto-service-lsstcomcam/values.yaml [hunk 7]
```diff
@@ -62,7 +62,7 @@ prompt-proto-service:
     # -- A factor by which to multiply `baseSize` for refcat datasets.
     refcatsPerImage: 4
     # -- A factor by which to multiply `baseSize` for templates and other patch-based datasets.
-    patchesPerImage: 4
+    patchesPerImage: 6
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values.yaml:65]: I don't know what's the optimal values, but I probably would start with those of LSSTComCamSim. Or where did you get these values that I probably missed?

File: applications/prompt-proto-service-lsstcomcam/values.yaml [hunk 8]
```diff
@@ -76,10 +76,6 @@ prompt-proto-service:
     # -- Set this to disable validation of S3 bucket names, allowing Ceph multi-tenant colon-separated names to be used.
     disableBucketValidation: 0
 
-  # -- The URI to a microservice that maps image metadata to a file location.
-  # If empty, Prompt Processing does not use a microservice.
-  raw_microservice: ""
-
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values.yaml:82]: Any reason to delete this?

File: applications/prompt-proto-service-lsstcomcam/values.yaml [hunk 9]
```diff
@@ -102,7 +98,6 @@ prompt-proto-service:
     # -- Server address for the alert stream
     server: "usdf-alert-stream-dev-broker-0.lsst.cloud:9094"
     # -- Topic name where alerts will be sent
-    # @default -- None, must be set
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values.yaml:105]: The None default seems good to me.  Any reason not?

File: applications/prompt-proto-service-lsstcomcam/values.yaml [hunk 10]
```diff
@@ -143,17 +138,14 @@ prompt-proto-service:
     memoryRequest: "2Gi"
     # -- The maximum memory limit for the full pod (see `containerConcurrency`).
     memoryLimit: "8Gi"
-    # -- To acommodate scheduling problems, Knative waits for a request for twice `worker.timeout`.
-    # This parameter adds extra time to that minimum (seconds).
-    extraTimeout: 10
```
Comment (hsinfang) [inline applications/prompt-proto-service-lsstcomcam/values.yaml:148]: Reason to delete this?

**Review by hsinfang — APPROVED** [review]

File: applications/prompt-proto-service-lsstcomcam/values.yaml (hunk 8 above)
Comment (erinleighh) [reply applications/prompt-proto-service-lsstcomcam/values.yaml:82]: Nope, that was an accident. Thanks for catching that!

File: applications/prompt-proto-service-lsstcomcam/values.yaml (hunk 7 above)
Comment (erinleighh) [reply applications/prompt-proto-service-lsstcomcam/values.yaml:65]: These values came from LATISS but I will update them to LSSTComCamSim's.

**Pull request merged**